zipline/pipeline/data/dataset.py

```diff
@@ -609,25 +608,24 @@ def __new__(cls, name, bases, dict_):
                 columns[k] = v
                 dict_[k] = _DataSetFamilyColumn(k)
 
-        is_base_class = bases == (cls._base_marker,)
-        if is_base_class:
-            bases = (object,)
+        is_abstract = dict_.pop('__abstract__', False)
```

Comment:
`__{}__` is reserved for the implementation of Python. I would assume this interacts with `ABCMeta, not part of our code. We should use something like `_abstract`.